### PR TITLE
final1: Expand input file tests

### PR DESCRIPTION
**Pull request by pcworld**
To test some special cases, I've modified `RecommendationSubtest#TASK_SHEET_INPUT_FILE`. We might want to split this up into separate tests. (Feel free to do this.)

### CODE REVIEW

**Comment by pcworld** [comment]
With 8b0bf3d2fe7796253041925c90f6880d315d7b15, it's probably difficult to tell in which iterations of the two for loops the test fails. Maybe exitTest should take an optional error message as a parameter.

**Comment by pcworld** [comment]
Maybe invalidArgumentCountTest should be merged into syntaxErrorsTest

**Comment by jGleitz** [comment]
I think we can merge this before fixing 8b0bf3d2fe7796253041925c90f6880d315d7b15 as this is development branch.

**Comment by jGleitz** [comment]
I like the for loop for the circle tests!

**Comment by jGleitz** [comment]
Do we use `PSEUDO_CIRCLE_INPUT_FILE1` and `PSEUDO_CIRCLE_INPUT_FILE2` in different subtests? If not, I'd suggest moving them to the subtests that's using it. If it only gets used by one method, we could even move it there.
If you plan on doing more with the input file we'll leave it there.

**Comment by jGleitz** [comment]
I'm strongly against modifying `RecommendationSubtest#TASK_SHEET_INPUT_FILE`. I think we should include the exact examples given on the task sheet.
Nevertheless, we should test for the spaces you inserted.

**Comment by pcworld** [comment]
> Do we use PSEUDO_CIRCLE_INPUT_FILE1 and PSEUDO_CIRCLE_INPUT_FILE2 in different subtests? If not, I'd suggest moving them to the subtests that's using it. If it only gets used by one method, we could even move it there.

I think at least one of them might be a good fit for testing `recommend` and `edges` command.
And I'm not sure if we should have a distinction between ValidInputFileTest and ValidNodesCommandTest, since if a test is valid we can test the commands anyway.

**Comment by jGleitz** [comment]
Alright, then we leave `PSEUDO_CIRCLE_INPUT_FILE1` and `PSEUDO_CIRCLE_INPUT_FILE2` were they are.

My thought was a separation of concerns. Users with multiple bugs may find it hard to debug based on our tests. Therefore I thought writing tests testing only one thing would makes sense. As soon as the user recognizes the tests fail because of his input parsing, he can debug until the tests for the input files go green. Afterwards, he continues to debug his `nodes` command.
Having a distinct test for the input file tells the user "Hey, there is one part of your program that actually works". Putting everything together might result in more frustration test results.

**Comment by pcworld** [comment]
> I'm strongly against modifying RecommendationSubtest#TASK_SHEET_INPUT_FILE. I think we should include the exact examples given on the task sheet.
> Nevertheless, we should test for the spaces you inserted.

Fixed in 1892dedb3bf5cb4bbab96a777f714cb7e5e3bc44 (not pretty but works, I guess)

> My thought was a separation of concerns. Users with multiple bugs may find it hard to debug based on our tests. Therefore I thought writing tests testing only one thing would makes sense. As soon as the user recognizes the tests fail because of his input parsing, he can debug until the tests for the input files go green. Afterwards, he continues to debug his nodes command.

This makes sense, however this could be more transparent from a developer perspective (i.e. you don't have to put a test in two places).
(But yes, getting tests done is more important right now, and I can live with a few extra seconds of copy&pasting.)

**Comment by jGleitz** [comment]
Fair enough. Merging.

> (not pretty but works, I guess)

We don't have to write the most beautiful code. It just has to remain maintainable. Which is provided, I guess.

File: src/final1/subtests/ValidNodesCommandTest.java [hunk 1]
```diff
@@ -14,14 +14,23 @@
...
-                "calc:202,centos5:105,centos6:106,centos7:107,impress:203,libreoffice:200,officesuite,operatingsystem,software,writer:201";
+        expectedResult = TASK_SHEET_NODES;
+        oneLineTest(addQuit(command), expectedResult, Input.getFile(TASK_SHEET_INPUT_FILE));
+    }
+    
+    @Test
+    public void spacesTest() {
+        command = "nodes";
+        expectedResult = TASK_SHEET_NODES;
         oneLineTest(addQuit(command), expectedResult, Input.getFile(TASK_SHEET_INPUT_FILE));
```
Comment (pcworld) [inline src/final1/subtests/ValidNodesCommandTest.java:34]: Should have been `TASK_SHEET_INPUT_FILE_SPACES`. Will fix with my next pull request

Comment (pcworld) [inline src/final1/subtests/ValidNodesCommandTest.java:34]: Fixed in 59382ad0035717d647488864c845272cc6a83f45

**Pull request merged**